src/api.rs

```diff
@@ -1680,41 +1686,46 @@ impl<'a> AuthenticatedApi<'a> {
             };
         }
 
-        match self
+        let region_url = match self
             .api
             .config
             .get_auth()
             .expect("auth should not be None for authenticated API!")
         {
-            Auth::Token(token) if token.payload().is_some() => {
-                let region_url = &token
-                    .payload()
-                    .expect("Payload already checked to have Some value")
-                    .region_url;
-
-                RegionSpecificApi {
-                    api: self,
-                    org,
-                    region_url: Some(region_url),
+            Auth::Token(token) => match token.payload() {
+                Some(payload) => Some(payload.region_url.clone().into()),
+                None => {
+                    let region_url = self.get_region_url(org);
+                    if let Err(err) = &region_url {
+                        log::warn!("Failed to get region URL due to following error: {err}");
```

Comment:
Perhaps, we should consider erroring here? I made this a warning for backwards compatibility, but since we only reach this line if we have verified the user is using `sentry.io`, we don't need to remain backwards compatible with self-hosted instances here, so an error might be okay. I could imagine that maybe some customers don't want us sending their data to the US region if we fail to retrieve the region URL.

Reply:
We can probably make this change in a separate PR, however